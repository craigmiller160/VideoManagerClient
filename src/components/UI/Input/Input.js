import React from 'react';
import PropTypes from 'prop-types';
import { Input as ReactInput, FormGroup, Label } from 'reactstrap';
import newid from '../../../utils/newid';
import classes from './Input.scss';

const Input = (props) => {
    const {
        label,
        type,
        name,
        onChange,
        value
    } = props;

    const id = newid();
    return (
        <FormGroup className={classes.Input}>
            <Label
                for={ id }
            >
                { label }
            </Label>
            <ReactInput
                id={ id }
                type={ type }
                name={ name }
                onChange={ onChange }
                value={ value }
            />
        </FormGroup>
    );
};

Input.propTypes = {
    type: PropTypes.string,
    label: PropTypes.string.isRequired,
    name: PropTypes.string,
    onChange: PropTypes.func,
    value: PropTypes.any
};

export default Input;