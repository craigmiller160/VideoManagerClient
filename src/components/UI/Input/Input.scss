.Input {
  label {
    font-weight: bold;
    margin: 0;
  }
}