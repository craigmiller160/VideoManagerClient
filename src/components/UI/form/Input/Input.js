import React from 'react';
import PropTypes from 'prop-types';
import StyledInput from '../../Styled/StyledInput';
import StyledLabel from '../../Styled/StyledLabel';
import newid from '../../../../utils/newid';
import createField from "../createField";
import StyledFormGroupDiv from '../../Styled/StyledFormGroupDiv';
import StyledTextArea from '../../Styled/StyledTextArea';

const Component = (props) => {
    if (props.type === 'textarea') {
        return (
            <StyledTextArea { ...props } />
        );
    }

    return (
        <StyledInput { ...props } />
    );
};

export const InputComponent = (props) => {
    const {
        label,
        type,
        input,
        textarea,
        inputProps
    } = props;

    const id = newid();

    // const component = 'textarea' === type ? StyledTextArea : StyledInput;

    return (
        <StyledFormGroupDiv hidden={ 'hidden' === type.toLowerCase() }>
            <StyledLabel
                htmlForm={ id }
            >
                { label }
            </StyledLabel>
            <Component
                id={ id }
                { ...input }
                { ...inputProps }
                type={ type }
                name={ input ? input.name : '' }
                rows={ textarea ? textarea.rows : null }
                cols={ textarea ? textarea.cols : null }
            />
        </StyledFormGroupDiv>
    );
};

InputComponent.defaultProps = {
    type: 'text',
    textarea: {
        resize: false
    },
    inputProps: {}
};

InputComponent.propTypes = {
    type: PropTypes.string,
    label: PropTypes.string,
    input: PropTypes.shape({
        name: PropTypes.string.isRequired
    }),
    inputProps: PropTypes.object,
    textarea: PropTypes.shape({
        rows: PropTypes.number,
        cols: PropTypes.number,
        resize: PropTypes.bool
    })
};

const Field = createField(InputComponent);
Field.propTypes = {
    type: PropTypes.string,
    label: PropTypes.string,
    name: PropTypes.string.isRequired
};

export default Field;