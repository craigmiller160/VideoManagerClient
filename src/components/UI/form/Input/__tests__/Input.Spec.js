import React from 'react';
import Input from '../Input';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Input', () => {
    const props = {
        label: 'My Input',
        type: 'text',
        name: 'MyInput'
    };

    it('should render correctly', () => {
        const component = shallow(
            <Input { ...props } />
        );
        expect(component).toMatchSnapshot();
    });

    it('should handle onChange', () => {
        let value = 'InitialValue';
        props.onChange = (newValue) => value = newValue;

        const component = shallow(
            <Input { ...props } />
        );

        const result = component.find('Input');
        expect(result.length).toEqual(1);

        result.simulate('change', {target: {value: 'My New Value'}});
        expect(value).toEqual({name: 'MyInput', value: 'My New Value'});
    });
});