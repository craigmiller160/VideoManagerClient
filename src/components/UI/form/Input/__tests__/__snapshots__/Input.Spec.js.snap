// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    label="My Input"
    name="MyInput"
    type="text"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "component": [Function],
      "label": "My Input",
      "name": "MyInput",
      "type": "text",
    },
    "ref": null,
    "rendered": null,
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "displayName": "Field",
      "render": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "component": [Function],
        "label": "My Input",
        "name": "MyInput",
        "type": "text",
      },
      "ref": null,
      "rendered": null,
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "displayName": "Field",
        "render": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
